parts/k8s/addons/cilium.yaml

```diff
@@ -1,69 +1,457 @@
+  # This may lead to policy drops or a change in loadbalancing decisions for a
+  # connection for some time. Endpoints may need to be recreated to restore
+  # connectivity.
+  #
+  # If this option is set to "false" during an upgrade from 1.3 or earlier to
+  # 1.4 or later, then it may cause one-time disruptions during the upgrade.
   preallocate-bpf-maps: "false"
+
+  # Regular expression matching compatible Istio sidecar istio-proxy
+  # container image names
   sidecar-istio-proxy-image: "cilium/istio_proxy"
-  tunnel: "vxlan"
+
+  # Encapsulation mode for communication between nodes
+  # Possible values:
+  #   - disabled
+  #   - vxlan (default)
+  #   - geneve
+  tunnel: disabled
+
+  # Name of the cluster. Only relevant when building a mesh of clusters.
   cluster-name: default
-  flannel-master-device: ""
-  flannel-uninstall-on-exit: "false"
-  flannel-manage-existing-containers: "false"
+
+  # DNS Polling periodically issues a DNS lookup for each `matchName` from
+  # cilium-agent. The result is used to regenerate endpoint policy.
+  # DNS lookups are repeated with an interval of 5 seconds, and are made for
+  # A(IPv4) and AAAA(IPv6) addresses. Should a lookup fail, the most recent IP
+  # data is used instead. An IP change will trigger a regeneration of the Cilium
+  # policy for each endpoint and increment the per cilium-agent policy
+  # repository revision.
+  #
+  # This option is disabled by default starting from version 1.4.x in favor
+  # of a more powerful DNS proxy-based implementation, see [0] for details.
+  # Enable this option if you want to use FQDN policies but do not want to use
+  # the DNS proxy.
+  #
+  # To ease upgrade, users may opt to set this option to "true".
+  # Otherwise please refer to the Upgrade Guide [1] which explains how to
+  # prepare policy rules for upgrade.
+  #
+  # [0] http://docs.cilium.io/en/stable/policy/language/#dns-based
+  # [1] http://docs.cilium.io/en/stable/install/upgrade/#changes-that-may-require-action
   tofqdns-enable-poller: "false"
+
+  # wait-bpf-mount makes init container wait until bpf filesystem is mounted
+  wait-bpf-mount: "false"
+  # Enable chaining with another CNI plugin
+  #
+  # Supported modes:
+  #  - none
+  #  - aws-cni
+  #  - flannel
+  #  - portmap (Enables HostPort support for Cilium)
+  cni-chaining-mode: generic-veth
+  # Disable the PodCIDR route to the cilium_host interface as it is not
+  # required. While chaining, it is the responsibility of the underlying plugin
+  # to enable routing.
+  enable-local-node-route: "false"
+
+  masquerade: "false"
+  enable-bpf-masquerade: "true"
+  enable-xt-socket-fallback: "true"
+  install-iptables-rules: "true"
+  auto-direct-node-routes: "false"
+  kube-proxy-replacement:  "probe"
+  node-port-bind-protection: "true"
+  enable-auto-protect-node-port-range: "true"
+  enable-session-affinity: "true"
+  k8s-require-ipv4-pod-cidr: "true"
```

Comment:
@errordeveloper I mean that if you change `k8s-require-ipv4-pod-cidr` to "false", that the CNI configuration is properly delivered by the cilium daemonset. But then kubelet complains thusly:

```
Sep 01 20:12:00 k8s-master-10101788-0 kubelet[1190]: W0901 20:12:00.478056    1190 cni.go:204] Error validating CNI config list {
Sep 01 20:12:00 k8s-master-10101788-0 kubelet[1190]:   "cniVersion": "0.3.0",
Sep 01 20:12:00 k8s-master-10101788-0 kubelet[1190]:   "name": "azure",
Sep 01 20:12:00 k8s-master-10101788-0 kubelet[1190]:   "plugins": [
Sep 01 20:12:00 k8s-master-10101788-0 kubelet[1190]:     {
Sep 01 20:12:00 k8s-master-10101788-0 kubelet[1190]:       "type": "azure-vnet",
Sep 01 20:12:00 k8s-master-10101788-0 kubelet[1190]:       "mode": "transparent",
Sep 01 20:12:00 k8s-master-10101788-0 kubelet[1190]:       "bridge": "azure0",
Sep 01 20:12:00 k8s-master-10101788-0 kubelet[1190]:       "ipam": {
Sep 01 20:12:00 k8s-master-10101788-0 kubelet[1190]:          "type": "azure-vnet-ipam"
Sep 01 20:12:00 k8s-master-10101788-0 kubelet[1190]:        }
Sep 01 20:12:00 k8s-master-10101788-0 kubelet[1190]:     },
Sep 01 20:12:00 k8s-master-10101788-0 kubelet[1190]:     {
Sep 01 20:12:00 k8s-master-10101788-0 kubelet[1190]:       "type": "portmap",
Sep 01 20:12:00 k8s-master-10101788-0 kubelet[1190]:       "capabilities": {"portMappings": true},
Sep 01 20:12:00 k8s-master-10101788-0 kubelet[1190]:       "snat": true
Sep 01 20:12:00 k8s-master-10101788-0 kubelet[1190]:     },
Sep 01 20:12:00 k8s-master-10101788-0 kubelet[1190]:     {
Sep 01 20:12:00 k8s-master-10101788-0 kubelet[1190]:        "name": "cilium",
Sep 01 20:12:00 k8s-master-10101788-0 kubelet[1190]:        "type": "cilium-cni"
Sep 01 20:12:00 k8s-master-10101788-0 kubelet[1190]:     }
Sep 01 20:12:00 k8s-master-10101788-0 kubelet[1190]:   ]
Sep 01 20:12:00 k8s-master-10101788-0 kubelet[1190]: }: [failed to find plugin "azure-vnet" in path [/opt/cni/bin]]
```

Reply:
Is `azure-vnet` installation somehow disabled by this configuration? What would it take ensure it's installed? Or is it in  a different path?